commands/deploy.go

```diff
@@ -329,6 +406,79 @@ func (p *deploycmd) updateRoute(c *cli.Context, appName string, ff *common.FuncF
 	return route.PutRoute(p.client, appName, ff.Path, rt)
 }
 
+func (p *deploycmd) updateFunction(c *cli.Context, appName string, ff *common.FuncFileV20180707) error {
+	fmt.Printf("Updating function %s using image %s...\n", ff.Name, ff.ImageNameV20180707())
+	fn := &modelsV2.Fn{}
+	if err := function.WithFuncFileV20180707(ff, fn); err != nil {
+		return fmt.Errorf("Error getting route with funcfile: %s", err)
+	}
+
+	app, err := apps.GetAppByName(appName)
+	if err != nil {
+		app = &models.App{
+			Name: appName,
+		}
+
+		err = apps.CreateApp(p.client, app)
+		if err != nil {
+			return err
+		}
+		app, err = apps.GetAppByName(appName)
+		if err != nil {
+			return err
+		}
+	}
+
+	fnRes, err := function.GetFnByName(p.clientV2, app.ID, ff.Name)
+	if err != nil {
```

Comment:
this seems too generic. If there's an error -> create? Can we be a bit more specific about the error that leads to a create? Not Found for instance.